### DISCUSSION TITLE
Get scrollarea's scrollbar styles

### DISCUSSION THREAD

**Discussion by Sajarin-M** [mantinedev/mantine#3152] [Q&A]
Can i get Mantine scroll area's scrollbar styles only and apply it to a normal div.

### Use case
I'am have a virtualized list in my project using [react-virtuoso](https://virtuoso.dev/)
there is a option to pass a [custom scroller](https://virtuoso.dev/custom-scroll-container/) to this list
passing `ScrollArea` component doesn't work as expected.
does anyone know how to do this?

**Comment by rmburg** [comment]
I used `react-window` for a virtualized list together with Mantine's ScrollArea. My code looks something like this:
```ts
import AutoSizer from "react-virtualized-auto-sizer";
import { FixedSizeList, ListChildComponentProps } from "react-window";
import { ScrollArea } from "@mantine/core";

// -- snip --

const listRef = useRef<FixedSizeList>(null);

// -- snip --

<div style={{ flexGrow: 1 }}>
  <AutoSizer>
    {({ height, width }) => (
      <ScrollArea
        style={{ width, height }}
        onScrollPositionChange={({ y }) => listRef.current?.scrollTo(y)}
      >
        <FixedSizeList
          height={height}
          width={width}
          style={{ overflow: "visible" }}
          itemCount={/* your data */.length}
          itemSize={/*item height in px*/}
          ref={listRef}
        >
          {({
            data,
            index,
            style,
          }: ListChildComponentProps<ItemDataType>) => (
            /* your row component*/
          )}
        </FixedSizeList>
      </ScrollArea>
    )}
  </AutoSizer>
</div>
```

**Reply by Sajarin-M** [reply]
Could you please share a code sandbox demo for this? It would be very helpful. Thank you very much.

**Reply by rmburg** [reply]
I have this one lying around from half a year ago, it's close to the code I pasted above.
https://codesandbox.io/s/react-window-with-scrollarea-g9dg6d?file=/src/App.tsx

**Reply by rmburg** [reply]
@Sajarin-M Please mark my answer as accepted if it answers your question.

**Reply by Sajarin-M** [reply]
@Narcha Thank you for your response, it was definitely helpful. however, my original question was specifically about achieving this behavior with react-virtuoso, and I was not able to get it to work using the solution you provided. I appreciate your input and would like to wait a bit longer to see if anyone else can offer a more suitable solution for `react-virtuoso` before marking your response as the answer. thank you again for your help.
